NeosModLoader/AssemblyHider.cs

```diff
@@ -86,12 +95,26 @@ private static bool IsModType(Type type)
 					// this is super weird, and probably shouldn't ever happen... but if it does, I want to know about it.
 					// since this is an edge case users may want to handle in different ways, the HideLateTypes nml config option allows them to choose.
 					bool hideLate = ModLoaderConfiguration.Get().HideLateTypes;
-					Logger.WarnInternal($"The \"{type}\" type does not appear to part of Neos or a mod. It is unclear whether it should be hidden or not. Due to the HideLateTypes config option being {hideLate} it will be {(hideLate ? "Hidden" : "Shown")}");
-					return hideLate; // hide the type only if hideLate == true
+					if (log)
+					{
+						Logger.WarnInternal($"The \"{name}\" {typeOrAssembly} does not appear to part of Neos or a mod. It is unclear whether it should be hidden or not. Due to the HideLateTypes config option being {hideLate} it will be {(hideLate ? "Hidden" : "Shown")}");
+					}
+					return hideLate && forceShowLate; // hide the thing only if hideLate == true
 				}
 			}
 		}
 
+		private static bool IsModAssembly(Assembly assembly, bool forceShowLate = false)
+		{
+			// this generates a lot of logspam, as a single call to AppDomain.GetAssemblies() calls this many times
+			return IsModAssembly(assembly, "assembly", assembly.ToString(), false, forceShowLate);
+		}
+
+		private static bool IsModType(Type type)
+		{
+			return IsModAssembly(type.Assembly, "type", type.ToString(), true, false);
```

Comment:
There's also quite a few parameters, which starts getting a bit messy keeping track of which one is which, would probably be clearer if named params are a available in net472? Or alternatively could change the input type to be an object, but iirc that wasn't as convenient in .NET...

Reply:
I've named the bools and added some more method docs in f24d079d